src/new-components/box/box.module.css

```diff
@@ -328,3 +328,45 @@
 .border-standard {
     border: 1px solid var(--reactist-framework-border);
 }
+
+/* text-align */
+.textAlign-start {
+    text-align: start;
+}
+.textAlign-center {
+    text-align: center;
+}
+.textAlign-end {
+    text-align: end;
+}
+.textAlign-justify {
+    text-align: justify;
+}
+@media (min-width: 768px /* --reactist-breakpoint-tablet */) {
+    .tablet-textAlign-start {
+        text-align: start;
+    }
+    .tablet-textAlign-center {
+        text-align: center;
+    }
+    .tablet-textAlign-end {
+        text-align: end;
+    }
+    .tablet-textAlign-justify {
+        text-align: justify;
+    }
+}
+@media (min-width: 992px /* --reactist-breakpoint-desktop */) {
+    .desktop-textAlign-start {
+        text-align: start;
+    }
+    .desktop-textAlign-center {
+        text-align: center;
+    }
+    .desktop-textAlign-end {
+        text-align: end;
+    }
+    .desktop-textAlign-justify {
+        text-align: justify;
+    }
+}
```

Comment:
Since everything ending in `textAlign-center` will always be aligned to the center and so forth, could we condense these styles by just having: 

```
[class*="textAlign-center"] {
  text-align: center;
}
```
..etc for center, end, justify and start. What do you think?

Reply:
Out of curiosity, why do we need the breakpoints? Would the expectation not be that anything containing `textAlign-center` is always centred regardless of the breakpoint? Unless there are some other styles that these are overriding

Reply:
This answers both of your comments:

Nope, what you suggest will actually break the system. Here's how it works:

The prefix `tablet-` or `desktop-` means that we only want that style to kick in when on screen sizes within the sizes of tablets or desktops.

Imagine a component like this:

```jsx
<Box textAlign={{ mobile: 'end', tablet: 'center', desktop: 'start' }} />
```

In HTML, it will end up being a div like this:

```html
<div className="box textAlign-end tablet-textAlign-center desktop-textAlign-start" />
```

So, if we do the styles of those classes like this:

```css
[class*="textAlign-start"] {
  text-align: start;
}
[class*="textAlign-center"] {
  text-align: center;
}
[class*="textAlign-end"] {
  text-align: end;
}
```

That div will always have its text aligned to the start or end (I'm not sure how the CSS rules are resolved when there are conflicting ones). But the fact is that all those styles will apply to it.

But what we really want is for certain styles to apply only on certain screen sizes. That's why the class names with the `tablet-` and `desktop-` prefixes are only applied inside a media query. The style for "mobile" does not have a prefix or a media query because this is the default style (however odd it may sound). How it works is: we apply the default style, which, if there are no overriding for tablet or desktop, will be the one that applies all the time. But, if there are overrides for tablet, for instance, this will effectively make the default class without a tablet/desktop prefix apply only on mobile screen sizes, because the override for tablet will kick in on any device larger than a mobile screen size.

Hope I answered your questions, and I did not misinterpreted what you meant.

Reply:
Hey @gnapse, thanks for the explanation, seeing it like this `<div className="box textAlign-end tablet-textAlign-center desktop-textAlign-start" />` makes a lot of sense why it wouldn't work!